### PR TITLE
Implemented riscv64 syscall support by modifying source code of opensbi and sbi-spec

### PR DESCRIPTION
**Pull request by Sssssaltyfish**
More precisely speaking, by having opensbi to pass unknown ecall to ruxos trap handler.
Note that due to the change of `[patch.crates.io]` in /Cargo.toml, CIs won't pass unless we init submodules before tests in CI procedures. Also, I couldn't find a `cfg(target_arch = "riscv64")` version of `[patch.crates.io]`.

Besides this modification, this PR does the following jobs:

1. opensbi and sbi-spec (used by sbi-rt used by ruxhal) are integrated via `git submodule`.

2. added 1000 to eid in sbi calls to distinguish sbi calls from posix sysacalls.

3. changed the default size of physical memory to 1G, as more would cause dtb page error in my environment.

### CODE REVIEW

File: platforms/riscv64-qemu-virt.toml
Comment (coolyjg) [inline platforms/riscv64-qemu-virt.toml:1]: A64 and X64 should do the same work too. The memory bug for R64 should be fixed before merging this PR.

**Comment by Sssssaltyfish** [comment]
As all tests are passed and I've successfully done a clean build under riscv64 in my environment, I guess this PR is ready to be merged. Please tell me if there's any improvement I can do before rebasing it into one commit.

File: modules/ruxhal/src/arch/riscv/trap.rs [hunk 1]
```diff
@@ -29,8 +29,23 @@ fn riscv_trap_handler(tf: &mut TrapFrame, _from_user: bool) {
...
+                    tf.regs.a3 as _,
+                    tf.regs.a4 as _,
+                    tf.regs.a5 as _,
+                ],
+            );
+            tf.regs.a0 = ret as _;
+        }
         _ => {
-            panic!(
+            error!(
```
Comment (coolyjg) [inline modules/ruxhal/src/arch/riscv/trap.rs:48]: It is better to just let it panic instead of throwing out an error.

**Review by coolyjg — COMMENTED** [review]
Just a minor suggestion. 
I think it is okay to merge. Please rebase into one commit.

File: modules/ruxhal/src/arch/riscv/trap.rs (hunk 1 above)
Comment (Sssssaltyfish) [reply modules/ruxhal/src/arch/riscv/trap.rs]: Sure. I forgot to change it back after debugging.

**Review by coolyjg — APPROVED** [review]

**Pull request merged**